api/v1beta1/disruption_types_test.go

```diff
@@ -137,3 +141,312 @@ var _ = Describe("Check if a target exist into DisruptionStatus targets list", f
 		})
 	})
 })
+
+var _ = Describe("Disruption", func() {
+
+	var (
+		defaultCreationTimestamp = time.Now()
+		notBeforeTime            = defaultCreationTimestamp.Add(time.Minute)
+	)
+
+	DescribeTable("TimeToInject", func(disruptionBuilder *builderstest.DisruptionBuilder, expectedTime time.Time) {
+		// Arrange
+		disruption := disruptionBuilder.WithCreationTime(defaultCreationTimestamp).Build()
+
+		// Action && Assert
+		Expect(disruption.TimeToInject()).To(Equal(expectedTime))
+	},
+		Entry(
+			"should return creationTimestamp if triggers is nil",
+			builderstest.NewDisruptionBuilder(), defaultCreationTimestamp),
+		Entry(
+			"should return triggers.createPods if triggers.inject is nil",
+			builderstest.NewDisruptionBuilder().WithDisruptionTriggers(DisruptionTriggers{
+				CreatePods: DisruptionTrigger{
+					NotBefore: metav1.NewTime(notBeforeTime),
+					Offset:    "",
+				},
+			}), notBeforeTime),
+		Entry(
+			"should return inject.notBefore if set",
+			builderstest.NewDisruptionBuilder().WithDisruptionTriggers(DisruptionTriggers{
+				Inject: DisruptionTrigger{
+					NotBefore: metav1.NewTime(notBeforeTime),
+					Offset:    "1",
```

Comment:
This should be a case where `Offset` is unset, no?
```suggestion
					Offset:    "",
```